src/codeu/chat/client/commandline/Chat.java

```diff
@@ -46,11 +46,19 @@ public Chat(Context context) {
   // the system wants to exit, the function will return false.
   //
   public boolean handleCommand(String line) {
-
+/*
     final Scanner tokens = new Scanner(line.trim());
-
     final String command = tokens.hasNext() ? tokens.next() : "";
-
+*/
+  final List<String> args = new ArrayList<>();
+  final Tokenizer tokenizer = new Tokenizer(line);
+  for(String token = tokenizer.next(); token != null; token = tokenizer.next())
+  {
+    args.add(token);
+  }
+  final String command = args.get(0);
+  args.remove(0);
+  
     // Because "exit" and "back" are applicable to every panel, handle
     // those commands here to avoid having to implement them for each
     // panel.
```

Comment:
I think you also have to change a few more parameters later down in the file to match the new code. Namely, `args` and `List<String>`.